src/datapoints-graph/datapoints-graph-view/chart-alarms.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { AlarmService, IAlarm, IFetchOptions, IResultList } from '@c8y/client';
+import { AlarmDetails } from '../alarm-event-selector';
+
+@Injectable()
+export class ChartAlarmsService {
+  constructor(private alarmService: AlarmService) {}
+
+  async listAlarms(params?, alarms?: AlarmDetails[]): Promise<IAlarm[]> {
+    const promises = alarms.map((alarm) => {
```

Comment:
if alarms param is optional, case for alarm to be undefined should be handled (right now it will throw 'cannot read map of undefined...'